Первое задание (Автоматизация Word)

EgorKorol4ov opened the pull request:
Вариант №13

EgorKorol4ov committed:
Выполнил первое задание(Титульный лист), загрузил файл, выгруженный программой

FlashAnton reviewed (changes requested):
Здравствуйте!

Необходимо обратить внимание и устранить следующие замечания:

1. Программный проект (Решение) назван незначимо. В сети можно найти миллион черновиков, названных "WindowsFormsApp1" и все они, преимущественно, черновики и пробы пера. Ваше же программное обеспечение предназначено для выполнения вполне конкретных задач.

2. Титульный лист, полученный в результате работы программного обеспечения, в целом, визуально и структурно не плох, но в деталях - "провалился". "Провалился" на этапе строгого следования утверждённым канонам известной печатной формы ("Титульный лист"), главным образом, при центрировании. Центрирование у Вас не является полноценным, а обладает первичным отступом по левому краю на 1,25 см. Но это уже не центрирование относительно границ страницы. Наименование вида работы (в Вашем случае - "Отчёт") должен на три-пять пунктов по шрифту преобладать над остальными надписями. И что-то Вы не рассчитали абзацные отступы - у Вас в результате генерации появляется вторая страница. Казалось бы, для рабочего документа, в котором будет происходить наполнение отчёта - удобно. А вот с точки зрения печати документа - нет. Принтер будет "выплёвывать" вторую чистую страницу (хорошо, что она без номера страницы, а то бы лист был испорчен). Подумайте, может быть переход от последнего рабочего абзаца к новому избыточен?

3. Не все интерфейсные элементы управления в Решении названы значимо с учётом оговоренной нотации с трёхбуквенным префиксом. Пускай, не с трёхбуквенным, ладно. Ваша нотация вида "button_" меня тоже вполне устраивает, но все интерфейсные элементы в Решении, без исключения, должны подчиняться тому же принципу наименования.

4. Не следует выполнять подобного рода обработку в разделе, который создан не для человека-разработчика, а для компоновки графического интерфейса графическим конструктором ("дизайнером" не человеком). Для этого предназначен другой раздел экранной формы. Не "дизайнер".
<img width="886" height="50" alt="image" src="https://github.com/user-attachments/assets/1df641ff-fc1a-448c-8568-baf14b2364d5" />
Эти действия приводят к тому, что графический конструктор перестаёт корректно функционировать.
<img width="1502" height="221" alt="image" src="https://github.com/user-attachments/assets/9f436a86-39ba-4ace-96ec-05c77496c83a" />
При этом с точки зрения компиляции Решения всё штатно и созданный Вами проект запускается, но зачем ломать сложившиеся каноны, предложенные ребятами из Microsoft?

5. Никаких var при объявлении переменных быть не должно. Язык со строгой типизаций, а потому нужно уделить время именно строгой типизации. Прошу отказаться от "ленивого" стиля разработки.

6. Титульный лист должен быть в одном Решении с шаблоном документа из индивидуального задания. Графический интерфейс должен предоставлять возможность переключения между настройками и выгрузкой титульного листа и выгрузкой шаблона документа согласно индивидуальному заданию.

При том, что эти части задания №1 "живут" в разных Решениях в репозитории, я, конечно же, проверю и в таком виде всё. Но в дальнейшем нужно будет выполнить агрегацию составленных алгоритмов в одно Решение.

На сегодня Ваша реализация при сравнении её с другими реализациями, направленными мне на проверку в последнее время, является наиболее близкой к учёту всех требований и ограничений. Она не вызывает отторжения, у неё есть авторское лицо и она вполне работоспособно. Всё вышеотмеченное - небольшие технические недочёты, которые просто необходимо подправить без тотального пересмотра архитектуры предложенного Решения.

EgorKorol4ov committed:
Переименовал решение

EgorKorol4ov committed:
Убрал создание второго листа

EgorKorol4ov committed:
Изменил название переменной

EgorKorol4ov committed:
Изменил отступы

EgorKorol4ov committed:
Объединил два пункта первого задания

EgorKorol4ov committed:
Пересоздал титульный лист с внесёнными изменениями

FlashAnton reviewed (changes requested):
Здравствуйте.

С объединением в одно решение - верно.

Но экранные формы должны приобрести корректные наименования в соответствии с выбранной нотацией. "btn" для кнопок вижу, соответственно, "frm" ожидается.

"var" неприемлемо. В решении всё должно быть строго типизировано. Не иначе.
<img width="203" height="29" alt="image" src="https://github.com/user-attachments/assets/202aafe5-b6fe-4293-91bd-cf20df2654ef" />

Если говорить про эргономичность размещения информации на экранной форме, то следует либо адаптировать надпись заголовка под имеющиеся размеры экранной формы (ввести разумные сокращения), либо адаптировать размеры формы под необходимую длину заголовка. И тот, и другой путь - корректны.

<img width="557" height="43" alt="image" src="https://github.com/user-attachments/assets/63d331a1-8df2-4044-ba28-5d16a8b43af0" />

Для экранной формы с настройками титульного листа с этим почти всё не плохо, то в экранной форме для генерации индивидуального документа - реализовано не лучшим образом, видимо, поспешно.

<img width="212" height="97" alt="image" src="https://github.com/user-attachments/assets/0e379a5e-7f1c-404f-91b7-7e0f6bfdfc27" />

Кривовато написано:
<img width="485" height="290" alt="image" src="https://github.com/user-attachments/assets/c4d80377-6627-49e7-9a2e-5284f68142d5" />

Во-первых, "var". Во-вторых, следовало бы составить метод с одним единственным return, расположенным в конце. Это было бы алгоритмически более строго. В целом, имеется архитектурный проигрыш. Можно было либо создавать экземпляр настроек считанных из файла, либо формировать экземпляр с настройками по умолчанию посредством двух перегрузок конструктора класса. Но я не прошу Вас менять архитектуру, если Вам такой формат удобен, я прошу привести в порядок алгоритмизацию. Всегда лучше иметь один общий вывод, чем несколько разных (если, конечно, речь не идёт о схеме переключения - там работают иные механизмы).

Ещё раз смотрите, что Вы делаете:

<img width="1482" height="286" alt="image" src="https://github.com/user-attachments/assets/632d79d8-7b76-4ef5-b680-d7f8d4746cd7" />

Не надо ломать графический конструктор.

Эта строка прекрасно имплементируется на этапе загрузки экранной формы и должна быть сформирована про мере загрузки экранной формы, а не на этапе её конструирования. Про жизненный цикл экранной формы побольше почитайте. Идейно проблема здесь в этом.

<img width="1093" height="40" alt="image" src="https://github.com/user-attachments/assets/2b8702db-7715-429a-a4a8-251859a7114b" />

К реализации компоновки документов Word замечаний не имею.

EgorKorol4ov committed:
Добавил frm.

Убрал все var.
Адаптировал надпись заголовка под размеры экранной формы для титульного листа.
Поправил форму для генерации индивидуального документа.
Переписал ParagraphSetting.
Исправил проблему, из-за которой не работал конструктор.

EgorKorol4ov committed:
Изменил название формы для индивидуального документа

FlashAnton reviewed (approved):
Здравствуйте.

Внесённые корректировки увидел.
Не везде они выполнены на 100% аккуратно, но приемлемо.
Достаточно для того, чтобы я согласился с выполненной Вами работой.

Готовьте сводный отчёт по практике.

Если он готов, то подсказывайте - где мне его смотреть.
